lib/exchange/provider/trocador_exchange_provider.dart

```diff
@@ -147,7 +124,16 @@ class TrocadorExchangeProvider extends ExchangeProvider {
       final rateId = responseJSON['trade_id'] as String? ?? '';
 
       var quotes = responseJSON['quotes']['quotes'] as List;
-      _provider = quotes.map((quote) => quote['provider']).toList();
+      _provider = quotes
+          .where((quote) =>
+              providerStates.containsKey(quote['provider']) &&
+              providerStates[quote['provider']] == true)
```

Comment:
I believe it would be better to just skip the ones that are turned off, instead of checking the ones that are on
```suggestion
          .where((quote) =>
              !providerStates[quote['provider']] == false)
```